src/Monolog/Handler/SocketHandler.php

```diff
@@ -276,9 +301,35 @@ private function writeToSocket($data)
             if ($socketInfo['timed_out']) {
                 throw new \RuntimeException("Write timed-out");
             }
+
+            if ($this->writingIsTimedOut($sent)) {
+                throw new \RuntimeException("Write timed-out, no data sent for `{$this->writingTimeout}` seconds, probably we got disconnected (sent $sent of $length)");
+            }
         }
         if (!$this->isConnected() && $sent < $length) {
             throw new \RuntimeException("End-of-file reached, probably we got disconnected (sent $sent of $length)");
         }
     }
+
+    private function writingIsTimedOut($sent)
+    {
+        $writingTimeout = (int) floor($this->writingTimeout);
+        if (0 === $writingTimeout) {
+            return false;
+        }
+
+        if ($sent !== $this->lastSentBytes) {
+            $this->lastWritingAt = time();
+            $this->lastSentBytes = $sent;
```

Comment:
Shouldn't you `return false` here? If it seems like it wrote successfully I don't think we should do the time check?